### PR TITLE
Firmware factor out checking of direction

### PR DESCRIPTION
**Pull request by purdeaandrei**
Includes PR #658.

Please review only after PR #658 is merged.

### CODE REVIEW

**Comment by purdeaandrei** [comment]
rebased

**Comment by purdeaandrei** [comment]
Note that I used the `register` keyword to make it even better (see the two individual commits).
If that's not desirable we can have only the first one.

File: firmware/main.c [hunk 1]
```diff
@@ -408,14 +408,15 @@ uint16_t bitstream_idx;
 
 void handle_pending_usb_setup() {
   __xdata struct usb_req_setup *req = (__xdata struct usb_req_setup *)SETUPDAT;
+  register bool arg_read = (req->bmRequestType & USB_DIR_IN);
```
Comment (whitequark) [inline firmware/main.c:411]: Can we call this `req_dir_in`? Good otherwise.

Comment (purdeaandrei) [reply firmware/main.c]: Sure, done

**Review by whitequark — APPROVED** [review]

**Pull request merged**